app/models/one_signal_player.rb

```diff
@@ -24,4 +24,26 @@ class OneSignalPlayer < ApplicationRecord
   belongs_to :user
 
   enum platform: %i[web mobile]
+
+  def self.filter_player_ids(player_ids, notif_type)
+    filtered_player_ids = []
+
+    player_objects = OneSignalPlayer.includes(
+      user: :notification_settings
+    ).where(
+      id: player_ids
+    )
+
+    player_objects.each do |player|
```

Comment:
It's more Ruby-ish to do `each_with_object` or (more commonly) `select` and then `map` to do this task of filtering and then getting IDs